src/modules/cpu/kernel/transpose_generic.hpp

```diff
@@ -0,0 +1,482 @@
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.  IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+THE SOFTWARE.
+*/
+
+#include "rppdefs.h"
+#include "rpp_cpu_simd.hpp"
+#include "rpp_cpu_common.hpp"
+using namespace std;
+
+void compute_strides(Rpp32u *strides, Rpp32u *shape, Rpp32u nDim)
+{
+    if (nDim > 0)
+    {
+        uint64_t v = 1;
+        for (int i = nDim - 1; i > 0; i--)
+        {
+            strides[i] = v;
+            v *= shape[i];
+        }
+        strides[0] = v;
+    }
+}
+
+void increment_ndim_ptr(Rpp32f **dstPtr, Rpp32u nDim, Rpp32u increment)
+{
+    for(int i = 0; i < nDim; i++)
+        dstPtr[i] += increment;
+}
+
+void rpp_store16_f32_f32_channelwise(Rpp32f **dstPtr, __m128 *p)
+{
+    _mm_storeu_ps(dstPtr[0], p[0]);
+    _mm_storeu_ps(dstPtr[1], p[1]);
+    _mm_storeu_ps(dstPtr[2], p[2]);
+    _mm_storeu_ps(dstPtr[3], p[3]);
+}
+
+void rpp_transpose4x8_avx(__m256 *pSrc, __m128 *pDst)
+{
+  __m256 tmp0, tmp1, tmp2, tmp3;
+  tmp0 = _mm256_shuffle_ps(pSrc[0], pSrc[1], 0x44);
+  tmp2 = _mm256_shuffle_ps(pSrc[0], pSrc[1], 0xEE);
+  tmp1 = _mm256_shuffle_ps(pSrc[2], pSrc[3], 0x44);
+  tmp3 = _mm256_shuffle_ps(pSrc[2], pSrc[3], 0xEE);
+  pSrc[0] = _mm256_shuffle_ps(tmp0, tmp1, 0x88);
+  pSrc[1] = _mm256_shuffle_ps(tmp0, tmp1, 0xDD);
+  pSrc[2] = _mm256_shuffle_ps(tmp2, tmp3, 0x88);
+  pSrc[3] = _mm256_shuffle_ps(tmp2, tmp3, 0xDD);
+
+  pDst[0] = _mm256_castps256_ps128(pSrc[0]);
+  pDst[1] = _mm256_castps256_ps128(pSrc[1]);
+  pDst[2] = _mm256_castps256_ps128(pSrc[2]);
+  pDst[3] = _mm256_castps256_ps128(pSrc[3]);
+  pDst[4] = _mm256_extractf128_ps(pSrc[0], 1);
+  pDst[5] = _mm256_extractf128_ps(pSrc[1], 1);
+  pDst[6] = _mm256_extractf128_ps(pSrc[2], 1);
+  pDst[7] = _mm256_extractf128_ps(pSrc[3], 1);
+}
+
+void compute_2d_transpose(Rpp32f *srcPtrTemp, Rpp32f *dstPtrTemp, Rpp32u height, Rpp32u width, Rpp32u srcRowStride, Rpp32u dstRowStride)
+{
+    Rpp32u alignedRows = (height / 4) * 4;
+    Rpp32u alignedCols = (width / 8) * 8;
+    Rpp32u vectorIncrement = 8;
+    Rpp32u dstRowVectorStride = vectorIncrement * dstRowStride;
+
+    int i = 0;
+    for(; i < alignedRows; i += 4)
+    {
+        Rpp32s k = (i / 4);
+        Rpp32f *srcPtrRow[4], *dstPtrRow[8];
+
+        for(int j = 0; j < 4; j++)
+            srcPtrRow[j] = srcPtrTemp + (i + j) * srcRowStride;
+        for(int j = 0; j < 8; j++)
+            dstPtrRow[j] = dstPtrTemp + j * dstRowStride + k * 4;
+
+        Rpp32u vectorLoopCount = 0;
+        for(; vectorLoopCount < alignedCols; vectorLoopCount += vectorIncrement)
+        {
+            __m256 pSrc[4];
+            pSrc[0] = _mm256_loadu_ps(srcPtrRow[0]);
```

Comment:
Is there already a rpp_simd_load(rpp_load_...);  and a rpp_simd_store(rpp_store_...); available? If not we can appropriately create inside the rpp_cpu_simd.hpp